src/data/move.ts

```diff
@@ -2875,6 +2875,164 @@ export class StatStageChangeAttr extends MoveEffectAttr {
   }
 }
 
+/**
+ * Attribute used to determine the Biome/Terrain-based secondary effect of Secret Power
+ */
+export class SecretPowerAttr extends MoveEffectAttr {
+  constructor() {
+    super(false);
+  }
+
+  /**
+   * Used to determine if the move should apply a secondary effect based on Secret Power's 30% chance
+   * @param user
+   * @param target
+   * @param move
+   * @param args
+   * @returns `true` if the move should | `false` if the roll fails and the move should not
+   */
+  override canApply(user: Pokemon, target: Pokemon, move: Move, args?: any[]): boolean {
+    this.effectChanceOverride = move.chance;
+    const moveChance = this.getMoveChance(user, target, move, this.selfTarget);
+    if (moveChance < 0 || moveChance === 100 || user.randSeedInt(100) < moveChance) {
+      this.effectChanceOverride = 100; // effectChanceOverride used in the application of the actual secondary effect
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  /**
+   * Used to apply the secondary effect to the target Pokemon
+   * @param user
+   * @param target
+   * @param move
+   * @param args
+   * @returns `true` if a secondary effect is successfully applied | 'false' if not
```

Comment:
```suggestion
   * @returns `true` if a secondary effect is successfully applied
```
When a function returns `true`/`false`, you only need to document one of them since the other is automatically the opposite (unless the function is doing something really weird, in which case you'd still need to document both outcomes).